src/search/indexer.cc

```diff
@@ -38,7 +38,7 @@ StatusOr<FieldValueRetriever> FieldValueRetriever::Create(SearchOnDataType type,
     Hash db(storage, ns);
     std::string ns_key = db.AppendNamespacePrefix(key);
     HashMetadata metadata(false);
-    auto s = db.GetMetadata(ns_key, &metadata);
+    auto s = db.GetMetadata(Database::GetOptions{}, ns_key, &metadata);
```

Comment:
is it be better to add a method with default GetOption,? so that we can avoid changing many places with `Database::GetOptions{}`

Reply:
I thought of add a method with default GetOption, applying a rule implicitly might break the consistency of system, .i.e, we may tent to use `GetMetadata` in default function and forget to use snapshot when we should do like that?

Reply:
I feel good about it's also easy to break if don't understand why we need to use the snapshot in some places. But the current implementation is also good to me. To see if other folks have thoughts on this. @PragmaTwice @torwig

Reply:
Actually not only for `GetMetadata`, I think we should also make snapshot as a parameter in other methods. So we'd better more careful about this.

Reply:
Yeah, so I create a GetOption. we can do this step-by-step

Reply:
From the perspective that we use the same snapshot to retrieve the metadata and perform other read-related work down the code, the change LGTM.